File: docs/guides/csharp/src/saveObjectsPublicUser.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SaveObjectsPublicUser
{
  private static readonly List<Dictionary<string, object>> playlists = []; // Your records

  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
      await client.SaveObjectsAsync(
        "<YOUR_INDEX_NAME>",
        playlists,
        false,
        1000,
        new RequestOptionBuilder().AddExtraHeader("X-Algolia-User-ID", "*").Build()
      );
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/go/src/saveObjectsPublicUser.go
package main

import (
	"github.com/algolia/algoliasearch-client-go/v4/algolia/search"
)

func saveObjectsPublicUser() {
	playlists := []map[string]any{{ /* Your records */ }}

	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	_, err = client.SaveObjects(
		"<YOUR_INDEX_NAME>", playlists, search.WithWaitForTasks(false), search.WithBatchSize(1000), search.WithHeaderParam("X-Algolia-User-ID", "*"))
	if err != nil {
		panic(err)
	}
}
